diff --git a/src/Makefile b/src/Makefile
index 2ecf8e2..f5f8b3d 100755
--- a/src/Makefile
+++ b/src/Makefile
@@ -31,7 +31,7 @@ include $(RT28xx_DIR)/os/linux/config.mk
 RTMP_SRC_DIR = $(RT28xx_DIR)/RT$(MODULE)
 
 #PLATFORM: Target platform
-PLATFORM = PC
+#PLATFORM = PC
 #PLATFORM = SYNOLOGY
 #PLATFORM = RASPBERRYPI
 #PLATFORM = 5VT
@@ -70,6 +70,7 @@ PLATFORM = PC
 #PLATFORM = UBICOM_IPX8
 #PLATFORM = INTELP6
 #PLATFORM = MSTARTV
+PLATFORM = LibreElec
 
 #APSOC
 ifeq ($(MODULE),3050)
@@ -116,6 +117,13 @@ CROSS_COMPILE = /home/zack/spksrc/distrib/toolchains/5.1/x86_64-pc-linux-gnu/bin
 endif
 endif
 
+ifeq ($(PLATFORM),LibreElec)
+ARCH = arm
+LINUX_SRC =
+LINUX_SRC_MODULE =
+CROSS_COMPILE =
+endif
+
 ifeq ($(PLATFORM), RASPBERRYPI)
 ifeq ($(LINUX_SRC),)
 LINUX_SRC = /home/zack/rasp/linux
@@ -320,6 +328,7 @@ LINUX_SRC = $(RT28xx_DIR)/../../../../build_linux
 ifeq ($(CROSS_COMPILE),)
 CROSS_COMPILE=armv7a-mediatek451_001_vfp-linux-gnueabi-
 endif
+
 CC=$(CROSS_COMPILE)gcc
 $(warning =============================================)
 $(warning CC=$(CC) for wifi driver)
@@ -421,10 +430,14 @@ ifeq ($(PLATFORM),DM6446)
 else
 ifeq ($(PLATFORM),FREESCALE8377)
 	$(MAKE) ARCH=powerpc CROSS_COMPILE=$(CROSS_COMPILE) -C  $(LINUX_SRC) SUBDIRS=$(RT28xx_DIR)/os/linux modules
+else
+ifeq ($(PLATFORM),LibreElec)
+	$(MAKE) ARCH=arm CROSS_COMPILE=$(CROSS_COMPILE) -C $(LINUX_SRC) SUBDIRS=$(RT28xx_DIR)/os/linux modules
 else
 	$(MAKE) -C $(LINUX_SRC) SUBDIRS=$(RT28xx_DIR)/os/linux modules
 endif
 endif
+endif
 
 ifeq ($(OSABL),YES)
 	cp -f os/linux/Makefile.6.netif $(RT28xx_DIR)/os/linux/Makefile
diff --git a/src/os/linux/config.mk b/src/os/linux/config.mk
index 7ca9852..a13ab08 100755
--- a/src/os/linux/config.mk
+++ b/src/os/linux/config.mk
@@ -1179,6 +1179,11 @@ CFLAGS := -D__KERNEL__ -I$(LINUX_SRC)/include/asm-mips/mach-generic -I$(LINUX_SR
 export CFLAGS
 endif
 
+ifeq ($(PLATFORM),LibreElec)
+    EXTRA_CFLAGS := $(WFLAGS) -I$(RT28xx_DIR)/include
+    export EXTRA_CFLAGS
+endif
+
 ifeq ($(PLATFORM),SYNOLOGY)
     ifneq (,$(findstring 2.4,$(LINUX_SRC)))
 	# Linux 2.4
@@ -1205,4 +1210,4 @@ ifeq ($(PLATFORM),RASPBERRYPI)
 	# Linux 2.6
 	EXTRA_CFLAGS := $(WFLAGS) -DRASPBERRYPI
     endif
-endif
\ No newline at end of file
+endif
