diff --git a/client/Makefile b/client/Makefile
index 1dccc53..bf49c75 100644
--- a/client/Makefile
+++ b/client/Makefile
@@ -83,9 +83,8 @@ OBJ      += ../common/daemon.o player/coreaudio_player.o browseZeroConf/browse_b
 
 else
 
-CXX       = g++
 CXXFLAGS += -pthread -DHAS_VORBIS -DHAS_ALSA -DHAS_AVAHI -DHAS_DAEMON
-LDFLAGS  += -lrt -lasound -lvorbis -lavahi-client -lavahi-common -latomic
+LDFLAGS  += -lrt -lasound -lvorbis -lavahi-client -lavahi-common -logg
 OBJ      += ../common/daemon.o player/alsa_player.o browseZeroConf/browse_avahi.o
 
 endif
diff --git a/server/Makefile b/server/Makefile
index 1a4e917..e7dea66 100644
--- a/server/Makefile
+++ b/server/Makefile
@@ -88,9 +88,8 @@ OBJ      += ../common/daemon.o publishZeroConf/publish_bonjour.o
 
 else
 
-CXX       = g++
 CXXFLAGS += -DHAS_AVAHI -DHAS_DAEMON -pthread
-LDFLAGS  += -lrt -lavahi-client -lavahi-common -latomic
+LDFLAGS  += -lrt -lavahi-client -lavahi-common -logg
 OBJ      += ../common/daemon.o publishZeroConf/publish_avahi.o 
 
 endif
