diff -Naurp oscam-emu/CMakeLists.txt oscam-emu-patch/CMakeLists.txt
--- oscam-emu/CMakeLists.txt	2019-01-17 08:44:14.000000000 +0200
+++ oscam-emu-patch/CMakeLists.txt	2019-01-17 08:47:43.000000000 +0200
@@ -527,7 +527,7 @@ endif (NOT OSCamOperatingSystem MATCHES
 
 set (exe_name "oscam")
 add_executable (${exe_name} ${exe_srcs} ${exe_hdrs} ${softcamkey_data})
-target_link_libraries (${exe_name} ${csoscam} ${csmodules} ${csreaders} csctapi cscrypt minilzo ffdecsa)
+target_link_libraries (${exe_name} ${csoscam} ${csmodules} ${csreaders} csctapi cscrypt minilzo ffdecsa udev)
 if(HAVE_LIBRT AND HAVE_LIBUSB)
    if (LIBUSBDIR)
 	set (libusb_link "imp_libusb")
diff -Naurp oscam-emu/utils/CMakeLists.txt oscam-emu-patch/utils/CMakeLists.txt
--- oscam-emu/utils/CMakeLists.txt	2019-01-17 08:44:14.000000000 +0200
+++ oscam-emu-patch/utils/CMakeLists.txt	2019-01-17 08:48:48.000000000 +0200
@@ -12,7 +12,7 @@ file (GLOB all_srcs ${exe_srcs})
 
 set (util_name "list_smargo")
 add_executable (${util_name} ${exe_srcs} ${exe_hdrs})
-target_link_libraries (${util_name} ${libusb_link} ${rt_link} ${setupapi_link} ${ole32_link} ${shell32_link} ${pthread_link} ${dl_link})
+target_link_libraries (${util_name} ${libusb_link} ${rt_link} ${setupapi_link} ${ole32_link} ${shell32_link} ${pthread_link} ${dl_link} udev)
 #----------------------- printout resume -----------------------------
 
 message (STATUS "Utils: operating system: ${OSCamOperatingSystem}")
